File: py-test/fcstats.py
'''
Created on Jul 2, 2016

@author: BPLPatron
'''
import utils
from collections import Counter

pcnt = 0
colw = 13

fcstats = Counter()

fcstats2 = Counter()

def cpy():
    for k in fcstats.keys():
        fcstats2[k] = fcstats[k]

def chk():
    for k in fcstats.keys():
        if fcstats2[k] != fcstats[k]:
            cpy()
            return True
    return False

def clr():
    for k in fcstats.keys():
        fcstats[k] = 0
    cpy()

def rightJ(s, l):
    while len(s) < l:
        s = ' ' + s
    return s

def formatNumber(num):
    return "{:15,}".format(num)

def fixedD(num, l):
    return ('{:' + str(l) + ',}').format(num)

def fixedS(s, l):
    return ('{:>' + str(l) + '}').format(s)

hprinted = False

def printHeader():
    global pcnt
    global colw
    if pcnt % 1 == 0:
        utils.writedata('\r\n')
        utils.writedata('\r\n')
        for k in fcstats.keys():
            utils.writedata(fixedS(k, colw))
        utils.writedata('\r\n')
        for k in fcstats.keys():
            utils.writedata(fixedS('----', colw))
        # utils.writedata('\r\n');
    pcnt += 1

def oprint():
    global colw
    if chk():
        utils.writedata('\r\n')
        for k, v in sorted(fcstats.items()):
            utils.writedata(fixedS(k, colw))
            utils.writedata(': ')
            utils.writedata(fixedD(v, colw))
            utils.writedata('\n')

cpy()

File: js-tools/lib/fcstats.js
var utils = require("utils");

var pcnt = 0;
var colw = 12;

var fcstats = {
  dcpy: 0,
  ddel: 0,
  fcpy: 0,
  fdel: 0,
  bcpy: 0,
  bdel: 0,
};
module.exports.fcstats = fcstats;

var fcstats2 = {};

function cpy() {
  for (var k in fcstats) {
    fcstats2[k] = fcstats[k];
  }
}
module.exports.cpy = cpy;

function chk() {
  for (var k in fcstats) {
    if (fcstats2[k] != fcstats[k]) {
      cpy();
      return true;
    }
  }
  return false;
}
module.exports.chk = chk;

function clr() {
  for (var k in fcstats) {
    fcstats[k] = 0;
  }
  cpy();
}
module.exports.clr = clr;

function rightJ(str, len) {
  if (str.length < len) {
    for (; str.length < len; ) str = " " + str;
  }
  return str;
}

function fixedD(num, len) {
  var s = rightJ(formatNumber(num), len);
  return s;
}

function formatNumber(num) {
  return num.toLocaleString();
}

var hprinted = false;

function printHeader() {
  if (pcnt % 10 == 0) {
    utils.writedata("\r\n");
    utils.writedata("\r\n");
    for (var k in fcstats) {
      utils.writedata(rightJ(k, colw));
    }
    utils.writedata("\r\n");
    for (var k in fcstats) {
      utils.writedata(rightJ("----", colw));
    }
    // utils.writedata('\r\n');
  }
  pcnt++;
}

function print() {
  if (chk()) {
    printHeader();
    utils.writedata("\r\n");
    for (var k in fcstats) {
      utils.writedata(fixedD(fcstats[k], colw));
    }
  }
}
module.exports.print = print;
